eventcrypto.go

```diff
@@ -243,6 +243,19 @@ func VerifyEventSignatures(ctx context.Context, events []Event, keyRing JSONVeri
 		}
 		senderServer := ServerName(senderDomain)
 		domains[senderServer] = true
+		// Synapse requires that the event ID domain has a valid signature.
+		// https://github.com/matrix-org/synapse/blob/develop/synapse/event_auth.py#L98-L104
+		// There's a good reason for this. Surely.
```

Comment:
I think so you can't spoof event ID domains on event IDs?

Reply:
Yes, I imagine so.